src/ReflectionType.php

```diff
@@ -85,14 +85,14 @@ public static function convertToDisplayType(\ReflectionType $type)
             'int'  => 'integer',
             'bool' => 'boolean'
         ];
-        $displayType = $type->type;
+        $displayType = (string)$type;
         if (isset($typeMap[$displayType])) {
             $displayType = $typeMap[$displayType];
         }
 
         $displayType = ltrim($displayType, '\\');
 
-        if ($type->allowsNull()) {
+        if (method_exists($type, 'allowsNull') && $type->allowsNull()) {
```

Comment:
> I think this is catching the class definition from the polyfill, which had no $this->allowsNull() method.

Then I think you need to add stubs in the polyfill class for all 3 methods: http://php.net/manual/en/class.reflectiontype.php